certdb/ocspstapling/ocspstapling_test.go

```diff
@@ -0,0 +1,173 @@
+package ocspstapling
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/base64"
+	"github.com/cloudflare/cfssl/certdb"
+	"github.com/cloudflare/cfssl/certdb/sql"
+	"github.com/cloudflare/cfssl/certdb/testdb"
+	"github.com/cloudflare/cfssl/helpers"
+	"github.com/google/certificate-transparency/go"
+	"golang.org/x/crypto/ocsp"
+	"math/big"
+	"os"
+	"testing"
+)
+
+func TestStapleSCTList(t *testing.T) {
+	// Create a CA certificate
+	issuer, issuerPrivKey, err := makeCert(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a certificate for which to make an OCSP response
+	responderCert, _, err := makeCert(issuer)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create an OCSP response
+	template := ocsp.Response{ // TODO: populate
+		SerialNumber: responderCert.SerialNumber,
+		IssuerHash:   crypto.SHA256,
+		Status:       ocsp.Good,
+	}
+
+	respDER, err := ocsp.CreateResponse(issuer, responderCert, template, issuerPrivKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create an empty DB of OCSP responses
+	// TODO: Test non-sqlite db's
+	gopath := os.Getenv("GOPATH")
+	dbPath := gopath + "/src/github.com/cloudflare/cfssl/certdb/testdb/certstore_development.db"
+	testDB := sql.NewAccessor(testdb.SQLiteDB(dbPath))
+
+	// Store the OCSP response in the DB
+	respSN := responderCert.SerialNumber.Text(16)
+	testDB.InsertOCSP(certdb.OCSPRecord{
+		Serial: respSN,
+		Body:   base64.StdEncoding.EncodeToString(respDER),
+		//Expiry:,
+	})
+
+	// Insert an SCT for the OCSP response we created
+	var zeroSCT ct.SignedCertificateTimestamp
+	err = StapleSCTList(testDB, respSN, "", []ct.SignedCertificateTimestamp{zeroSCT}, issuer, issuerPrivKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the SCT was inserted
+	recs, err := testDB.GetOCSP(respSN, "")
+	if len(recs) == 0 {
+		t.Fatal("SCT could not be retrieved from DB:", zeroSCT)
+	}
+
+	respDER, err = base64.StdEncoding.DecodeString(recs[0].Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	response, err := ocsp.ParseResponse(respDER, issuer)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	scts := SCTListFromOCSPResponse(response)
+	if len(scts) == 0 {
+		t.Fatal("No SCTs in OCSP response:", response)
+	}
+
+	// TODO: Verify that scts[0] is equivalent to zeroSCT
+}
+
+// TODO: COMMENT
+var serialCounter int64
+
+// TODO: COMMENT
+func nextSN() *big.Int {
+	i := big.NewInt(serialCounter)
+	serialCounter++
+	return i
+}
+
+// TODO: COMMENT
```

Comment:
The comment is optional, given the name and return signature. It shouldn't be left as a TODO, though.